index.js

```diff
@@ -209,13 +210,15 @@ module.exports = async function (url, options) {
 
                 let timeoutHandler;
                 if (retryOptions.socketTimeout) {
-                    const controller = new AbortController();
+                    const abortCon = options.AbortController || AbortController;
+                    const controller = new abortCon();
                     timeoutHandler = setTimeout(() => controller.abort(), retryOptions.socketTimeout);
                     options.signal = controller.signal;
                 }                
     
                 try {
-                    const response = await fetch(url, options);
+                    const fetchFn = options.fetch || fetch;
```

Comment:
this recursively invokes  `wrappedFetch();` so i would prefer for you to set the `fetchFn` once if possible. (somehow globally maybe).
Maybe there is a way to declare on import which fetch to use?

In case the options may get wiped away between retries (maybe this is too unrealistic)